### PR TITLE
Distinguish between eviction and successful restart in e2e

### PR DESCRIPTION
**Pull request by bskiba**

### CODE REVIEW

**Comment by bskiba** [comment]
@kgolab

**Comment by bskiba** [comment]
@kgolab

File: vertical-pod-autoscaler/e2e/common.go [hunk 1]
```diff
@@ -227,15 +227,43 @@ func WaitForPodsRestarted(f *framework.Framework, podList *apiv1.PodList) error
...
+		return false
+	}
+	evictedCount := GetEvictedPodsCount(currentPodSet, initialPodSet)
+	framework.Logf("%v of initial pods were already evicted", evictedCount)
+	return evictedCount > 0
 }
 
-// GetRestartedPodsCount returns the count of pods from initialPodSet that have
+// GetEvictedPodsCount returns the count of pods from initialPodSet that have
 // been restarted comparing to currentPodSet.
```
Comment (kgolab) [inline vertical-pod-autoscaler/e2e/common.go]: nit: I guess it's evicted.

File: vertical-pod-autoscaler/e2e/updater.go [hunk 2]
```diff
@@ -110,7 +110,7 @@ func testRestartsPods(f *framework.Framework, controllerKind string) {
 	SetupVPA(f, "200m", vpa_types.UpdateModeAuto)
 
 	ginkgo.By("Waiting for pods to be restarted")
```
Comment (kgolab) [inline vertical-pod-autoscaler/e2e/updater.go]: nit: evicted?

File: vertical-pod-autoscaler/e2e/updater.go [hunk 3]
```diff
@@ -94,7 +94,7 @@ var _ = UpdaterE2eDescribe("Updater", func() {
 		ginkgo.By("Checking enough pods were restarted.")
```
Comment (kgolab) [inline vertical-pod-autoscaler/e2e/updater.go]: nit: here & two lines earlier it's "evicted" again.

**Comment by kgolab** [comment]
Mostly nits in the comments but all together they hint at re-checking whole file for distinction between evicted & restarted - I've found at least one place where I think we are OK with "no evictions" instead of "no restarts".

**Comment by kgolab** [comment]
Thanks, LGTM now.

**Pull request merged**